### PR TITLE
Adds docs for location customizers

### PR DESCRIPTION
**Pull request by aledsage**

### CODE REVIEW

File: docs/guide/ops/locations/location-customizers.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+title: Location customizers
+layout: website-normal
+---
+
+Apache Brooklyn supports a number of ways to configure and customize locations. These include
+the `JcloudsLocationCustomizer`, which is for advanced customization of VM provisioning through jclouds.
+There is also a `MachineLocationCustomizer`, which allows customization of machines being obtained 
+from any kind of location (including Bring Your Own Nodes).
```
Comment (sjcorbett) [inline docs/guide/ops/locations/location-customizers.md]: Could link "Bring Your Own Nodes" to https://brooklyn.incubator.apache.org/v/latest/ops/locations/#byon.

File: docs/guide/ops/locations/location-customizers.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  This is used to influence the choice of VM image, hardware profile, etc. This hook is not normally
+  required as the location configuration options can be used in instead.
+
+* Customize the `org.jclouds.compute.domain.Template`, to be used when creating the machine. This  
+  hook is most often used for performing custom actions - for example to create or modify a security 
+  group or volume, and to update the template's options to use that.
+
+* Customize the `org.jclouds.compute.options.TemplateOptions` to be used when creating the machine.
+  The `TemplateOptions` could be cast to a cloud-specific sub-type (if this does not have to work
+  across different clouds). Where the use-cases is to just set simple configuration on the
```
Comment (sjcorbett) [inline docs/guide/ops/locations/location-customizers.md]: `use-cases` should be singular.

File: docs/guide/ops/locations/location-customizers.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+* Post-release of the machine (i.e. after asking jclouds to destroying the machine).
+
+To register a `JcloudsLocationCustomizer` in YAML, the config key `customizers` can be used to 
+provide a list of instances. Each instance can be defined using `$brooklyn:object` to indicate 
+the type and its configuration. For example:
+
+    location:
+      jclouds:aws-ec2:us-east-1:
+        cutomizers:
```
Comment (sjcorbett) [inline docs/guide/ops/locations/location-customizers.md]: Typo: `cutomizers`

File: docs/guide/ops/locations/location-customizers.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        - $brooklyn:object:
+            type: com.acme.brooklyn.MyJcloudsLocationCustomizer
+
+To register `JcloudsLocationCustomizer` instances programmatically, set the config key
+`JcloudsLocationConfig.JCLOUDS_LOCATION_CUSTOMIZERS` on the location, or pass this 
+config option when calling `location.obtain(options).
```
Comment (sjcorbett) [inline docs/guide/ops/locations/location-customizers.md]: Missing the closing backtick.

File: docs/guide/ops/locations/location-customizers.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+    location:
+      jclouds:aws-ec2:us-east-1:
+        machineCustomizers:
+        - $brooklyn:object:
+            type: com.acme.brooklyn.MyMachineLocationCustomizer
+
+To register `MachineLocationCustomizer` instances programmatically, set the config key
+`CloudLocationConfig.MACHINE_LOCATION_CUSTOMIZERS` on the location, or pass this 
+config option when calling `location.obtain(options).
```
Comment (sjcorbett) [inline docs/guide/ops/locations/location-customizers.md]: Missing closing backtick.

**Comment by aledsage** [comment]
Thanks @sjcorbett (and thanks also @andreaturli for reviewing) - I've incorporate those comments.

Merging now.

**Pull request merged**